content/blog/2023/09/25/2023-09-25-building-jenkinsio-with-alternative-tools.adoc

```diff
@@ -0,0 +1,72 @@
+---
+:layout: post
+:title: "GSoC Building Jenkins.io with alternative tools"
+:tags:
+- gsoc
+- gsoc2023
+- jenkins.io
+- jenkins-infra
+:author: vandit1604
+:sig: gsoc
+description: >
+  The Building Jenkins.io with Alternative Tools is a project under Google Summer of Code (GSoC) 2023.
+opengraph:
+  image: /images/gsoc/opengraph.png
+links:
+  gitter: https://matrix.to/#/#jenkinsci_gsoc-2023-building-jenkinsio:matrix.org
+---
+
+This blog showcases all the work done in the project link:/projects/gsoc/2023/projects/alternative-jenkinsio-build-tool/[Building Jenkins.io with Alternative Tools] during Google Summer of Code 2023.
+
+== Table of Content
+
+. <<About Project>>
+. Coding <<Phase 1>>
+. Coding <<Phase 2>>
+. <<Future Improvements>>
+. <<Acknowledgments>>
+. <<Useful Links>>
+
+== About Project
+
+Jenkins.io is generated as a static website using Awestruct from AsciiDoc sources. One of the drawbacks of the current build method is that the technical documentation is not Versioned. It is thus not possible to view the documentation for a given Jenkins version. Only the latest can be viewed. This can lead to unnecessary confusion and is a worse experience than many other documentation sites like the git site, FreeBSD, and others.. Other drawback of Awestruct is that it is greatly under maintained since last two years and people from jenkins-infra are currently maintaining it. Relying on awestruct is not a good thing for a longer run. 
```

Comment:
We generally prefer sentence per line formatting so that code review is easier.

I replaced the FreeBSD example with Python because I was unable to find a way to show multiple versions of the the current FreeBSD documentation.  They state that previous versions are available, but I did not find a link that would show previous versions of the FreeBSD documentation.

```suggestion
The link:/[Jenkins documentation site] is generated as a static website using Awestruct from AsciiDoc sources.
One of the drawbacks of the current build method is that the technical documentation is not versioned.
It is thus not possible to view the documentation for a given Jenkins version.
Only the latest can be viewed.
This can lead to unnecessary confusion and is a worse experience than many other documentation sites like the link:https://git-scm.com/docs/git[git site], link:https://docs.python.org/3.11/[Python[, and others.
Another drawback of Awestruct is that it is greatly under maintained in the last two years and people from jenkins-infra are currently maintaining it.
Relying on awestruct is not a good thing in the long run. 
```